--- a/qmi_wwan_f.c
+++ b/qmi_wwan_f.c
@@ -1208,8 +1208,12 @@ static int qmap_register_device(sQmiWwan
 
     priv->agg_skb = NULL;
     priv->agg_count = 0;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,15,0)
     hrtimer_init(&priv->agg_hrtimer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
     priv->agg_hrtimer.function = rmnet_usb_tx_agg_timer_cb;
+#else
+    hrtimer_setup(&priv->agg_hrtimer, rmnet_usb_tx_agg_timer_cb, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
+#endif
     INIT_WORK(&priv->agg_wq, rmnet_usb_tx_agg_work);
     ktime_get_ts64(&priv->agg_time);
     spin_lock_init(&priv->agg_lock);
